lib/phoenix_component.ex

```diff
@@ -660,102 +866,102 @@ defmodule Phoenix.Component do
   information is mostly used for documentation and reflection purposes.
 
   On the side of the LiveView component itself, defining attributes provides
   the following quality of life improvements:
 
-    * The default value of all attributes will be added to the `assigns`
-      map upfront
-
-    * Required struct types are annotated and emit compilation warnings.
-      For example, if you specify `attr :user, User, required: true` and
-      then you write `@user.non_valid_field` in your template, a warning
-      will be emitted
+  * The default value of all attributes will be added to the `assigns`
+    map upfront. Note that unless an attribute is marked as required 
+    or has a default defined, omitting a value for an attribute will 
+    result in `nil` being passed as the default value to the `assigns`
+    map, regardless of the type defined for the attribute.
+      
+  * Attribute documentation is generated for the component.
 
-  This list may increase in the future.
+  * Required struct types are annotated and emit compilation warnings.
+    For example, if you specify `attr :user, User, required: true` and
+    then you write `@user.non_valid_field` in your template, a warning
+    will be emitted.
+    
+  * Calls made to the component are tracked for reflection and 
+    validation purposes.
 
-  ## Documentation
+  ## Documentation Generation
 
   Public function components that define attributes will have their attribute
   types and docs injected into the function's documentation, depending on the
   value of the `@doc` module attribute:
 
-    * if `@doc` is a string, the attribute docs are injected into that string.
-      The optional placeholder `[[INJECT LVDOCS]]` can be used to specify where
-      in the string the docs are injected. Otherwise, the docs are appended
-      to the end of the `@doc` string.
+  * if `@doc` is a string, the attribute docs are injected into that string.
+    The optional placeholder `[[INJECT LVDOCS]]` can be used to specify where
+    in the string the docs are injected. Otherwise, the docs are appended
+    to the end of the `@doc` string.
 
-    * if `@doc` is unspecified, the attribute docs are used as the
-      default `@doc` string.
+  * if `@doc` is unspecified, the attribute docs are used as the
+    default `@doc` string.
 
-    * if `@doc` is false, the attribute docs are omitted entirely.
+  * if `@doc` is `false`, the attribute docs are omitted entirely.
 
   The injected attribute docs are formatted as a markdown list:
 
-    ```markdown
-    * `name` (`:type`) (required) - attr docs. Defaults to `:default`.
-    ```
+  ```markdown
+  * `name` (`:type`) (required) - attr docs. Defaults to `:default`.
+  ```
 
   By default, all attributes will have their types and docs injected into
   the function `@doc` string. To hide a specific attribute, you can set
   the value of `:doc` to `false`.
-
-  ## Examples
```

Comment:
I think we should keep the examples here too, in case someone jumps to the documentation of the function and wants a quick look without reading all of module docs.